atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -1427,9 +1430,124 @@ public Void apply(Client client) throws Exception {
+                @Override
+                public Void apply(Client client) throws Exception {
+                    Cell globalDdlLockCell = Cell.create(CassandraConstants.GLOBAL_DDL_LOCK.getBytes(), CassandraConstants.GLOBAL_DDL_LOCK_COLUMN_NAME.getBytes());
+                    ByteBuffer rowName = ByteBuffer.wrap(globalDdlLockCell.getRowName());
+                    Column ourUpdate = lockColumnWithValue(Longs.toByteArray(perOperationNodeIdentifier));
+
+                    List<Column> expected = ImmutableList.of(lockColumnWithValue(Longs.toByteArray(CassandraConstants.GLOBAL_DDL_LOCK_CLEARED_VALUE)));
+
+                    CASResult casResult = writeLockWithCAS(client, rowName, expected, ourUpdate);
+
+                    int timesAttempted = 0;
+                    long timeSlept = 0;
+
+                    while (!casResult.isSuccess()) { // could have a timeout controlling this level, confusing for users to set both timeouts though
+                        if (casResult.getCurrent_valuesSize() == 0) { // never has been an existing lock
+                            // special case, no one has ever made a lock ever before
+                            // this becomes analogous to putUnlessExists now
+                            expected = ImmutableList.<Column>of();
+                        } else {
+                            Column existingValue = Iterables.getOnlyElement(casResult.getCurrent_values(), null);
+                            if (existingValue == null) {
+                                throw new IllegalStateException("Something is wrong with underlying locks. Consult support for guidance on manually examining and clearing locks from " + CassandraConstants.LOCK_TABLE + " table.");
+                            }
+                            expected = ImmutableList.of(lockColumnWithValue(Longs.toByteArray(CassandraConstants.GLOBAL_DDL_LOCK_CLEARED_VALUE)));
+                        }
+
+                        if (timeSlept > configManager.getConfig().schemaMutationTimeoutMillis() * 4) { // possibly dead remote locker
+                            throw new TimeoutException(String.format("We have timed out waiting on the current schema mutation lock holder.  " +
+                                    "We have tried to grab the lock for %d milliseconds unsuccessfully.  Please try restarting the AtlasDB client." +
+                                    "If this occurs repeatedly it may indicate that the current lock holder has died without releasing the lock." +
+                                    "This will require manual intervention to repair, please contact support.", timeSlept));
+                        }
+
+
+                        long timeToSleep = CassandraConstants.TIME_BETWEEN_LOCK_ATTEMPT_ROUNDS_MILLIS * (long) Math.pow(2, timesAttempted++);
+                        Thread.sleep(timeToSleep);
+                        timeSlept += timeToSleep;
```

Comment:
might as well make this an actual timer, as some of the operations here actually can take a significant amount of time (default CAS contention timeout is 5 seconds)